Commands/process.go

```diff
@@ -0,0 +1,54 @@
+package Commands
+
+import (
+	"errors"
+	"os"
+	"regexp"
+	"strings"
+
+	"github.com/slack-go/slack"
+)
+
+func ProcessMessage(channel string, msgTimestamp string, msg string) error {
+
+	botID := "<@U025MPQCB9A> "
+
+	name, err := parseCommandName(strings.TrimPrefix(msg, botID))
+	if err != nil {
+		return err
+	}
+
+	response, err := RunCommand(name)
+	replyWith(channel, msgTimestamp, response)
+
+	return err
+}
+
+func parseCommandName(msg string) (string, error) {
+	commands := map[string]string{
+		"balance":  `^my balance`,
+		"transfer": `^transfer to`,
+	}
+	for name, pattern := range commands {
+		matched, err := regexp.MatchString(pattern, msg)
+		if err != nil {
+			return "", nil
+		}
+		if matched {
+			return name, nil
+		}
+	}
+	return "", errors.New("no match")
```

Comment:
Using the builtin `errors` package is totally fine, but one of the annoyances of Go is that it doesn't give you stack traces. When this crashes you'll get a `no match` error message with no location information. In this case, it'll be fine, you can grep and find this location, but it gets boring pretty fast when the error messages are less unique. You might want to use [github.com/pkg/errors](https://pkg.go.dev/github.com/pkg/errors) which provide builtin stack traces (and you can use `errors.WithStack(err)` to add a stack trace to an error you receive from a package that doesn't provide them by default.